SimpleWAWS/Code/TemplatesManager.cs

```diff
@@ -60,35 +64,72 @@ static TemplatesManager()
                     AppService = AppService.Logic,
-                    CsmTemplateFilePath = HostingEnvironment.MapPath("~/CSMTemplates/PingSite.json"),
-                    Description = Resources.Server.Templates_PingSiteDescription
+                    CsmTemplateFilePath = HostingEnvironment.MapPath("~/ARMTemplates/PingSite.json"),
+                    Description = Resources.Server.Templates_PingSiteDescription,
+                    MSDeployPackageUrl = $"https://github.com/fashaikh/appservice-zipped-templates/raw/master/Logic/Ping%20Site.zip"
                 });
                 list.Add(new JenkinsTemplate
                 {
                     Name = "Jenkins CI",
                     SpriteName = "sprite-PingSite PingSite",
                     AppService = AppService.Jenkins,
-                    CsmTemplateFilePath = HostingEnvironment.MapPath("~/CSMTemplates/JenkinsResource.json"),
+                    CsmTemplateFilePath = HostingEnvironment.MapPath("~/ARMTemplates/JenkinsResource.json"),
                     Description = Resources.Server.Templates_JenkinsDescription
                 });
+                _baseARMTemplate = JObject.Parse(File.ReadAllText(HostingEnvironment.MapPath("~/ARMTemplates/BaseARMTemplate.json"),  System.Text.Encoding.ASCII)
+                    .Replace(Environment.NewLine, String.Empty)
+                    .Replace(@"\", String.Empty));
                 //TODO: Implement a FileSystemWatcher for changes in the directory
             }
-            catch (Exception)
+            catch (Exception ex)
             {
                 _templatesList = Enumerable.Empty<BaseTemplate>();
             }
         }
 
+        private static object getShortName(string templateName)
+        {
+            return templateName.Replace(" ", "").Replace("#", "Sharp").Replace(".", "").Replace("+", "");
+        }
+
         public static IEnumerable<BaseTemplate> GetTemplates()
         {
             return _templatesList;
         }
+        public static JObject GetARMTemplate(BaseTemplate template)
+        {
+            var armTemplate = _baseARMTemplate;
+            updateParameters(ref armTemplate, template);
+            //TODO: Add mroe specific app settings as needed.
+            updateAppSettings(ref armTemplate, template);
+            updateConfig(ref armTemplate, template);
+            return armTemplate;
+        }
+        private static void updateParameters(ref dynamic temp, BaseTemplate template)
+        {
+            var shortName = getShortName(template.Name);
+            temp.parameters.appServiceName.defaultValue = string.Concat("My-", shortName, "App-", Guid.NewGuid().ToString().Split('-')[0]);
+            temp.parameters.msdeployPackageUrl.defaultValue = template.MSDeployPackageUrl;
+        }
+
+        private static void updateConfig(ref dynamic temp, BaseTemplate template)
+        {
+            temp.resources[1].resources[0].properties.templateName = template.Name ;
```

Comment:
this seems to be to bake too much knowledge of a dynamic object structure in the code. isn't there a way to do this with static type validation?